Index: sagemaker-code-editor/vscode/src/vs/platform/product/common/marketplace.ts
===================================================================
--- /dev/null
+++ sagemaker-code-editor/vscode/src/vs/platform/product/common/marketplace.ts
@@ -0,0 +1,26 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Coder Technologies. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { IProductConfiguration } from 'vs/base/common/product';
+
+/**
+ * Allows for overriding the extension gallery configuration via the
+ * `'EXTENSIONS_GALLERY'` environment variable.
+ *
+ * @example
+ * ```sh
+ * export EXTENSIONS_GALLERY='{"serviceUrl": "https://extensions.coder.com/api"}'
+ * ```
+ */
+export function parseExtensionsGalleryEnv(extensionsGalleryEnv: string): IProductConfiguration['extensionsGallery'] {
+return {
+    serviceUrl: '',
+    itemUrl: '',
+    resourceUrlTemplate: '',
+    controlUrl: '',
+    recommendationsUrl: '',
+    ...JSON.parse(extensionsGalleryEnv),
+};
+}
\ No newline at end of file
Index: sagemaker-code-editor/vscode/src/vs/platform/product/common/product.ts
===================================================================
--- sagemaker-code-editor.orig/vscode/src/vs/platform/product/common/product.ts
+++ sagemaker-code-editor/vscode/src/vs/platform/product/common/product.ts
@@ -6,6 +6,7 @@
 import { env } from 'vs/base/common/process';
 import { IProductConfiguration } from 'vs/base/common/product';
 import { ISandboxConfiguration } from 'vs/base/parts/sandbox/common/sandboxTypes';
+import { parseExtensionsGalleryEnv } from 'vs/platform/product/common/marketplace';
 
 /**
  * @deprecated You MUST use `IProductService` if possible.
Index: sagemaker-code-editor/vscode/src/vs/workbench/contrib/extensions/browser/extensionsViewlet.ts
===================================================================
--- sagemaker-code-editor.orig/vscode/src/vs/workbench/contrib/extensions/browser/extensionsViewlet.ts
+++ sagemaker-code-editor/vscode/src/vs/workbench/contrib/extensions/browser/extensionsViewlet.ts
@@ -64,6 +64,8 @@ import { ILocalizedString } from 'vs/pla
 import { registerNavigableContainer } from 'vs/workbench/browser/actions/widgetNavigationCommands';
 import { MenuWorkbenchToolBar } from 'vs/platform/actions/browser/toolbar';
 import { createActionViewItem } from 'vs/platform/actions/browser/menuEntryActionViewItem';
+import { IProductService } from 'vs/platform/product/common/productService';
+import { memoize } from 'vs/base/common/decorators';
 
 export const DefaultViewsContext = new RawContextKey<boolean>('defaultExtensionViews', true);
 export const ExtensionsSortByContext = new RawContextKey<string>('extensionsSortByValue', '');
@@ -87,19 +89,22 @@ const SortByUpdateDateContext = new RawC
 const REMOTE_CATEGORY: ILocalizedString = localize2({ key: 'remote', comment: ['Remote as in remote machine'] }, "Remote");
 
 export class ExtensionsViewletViewsContribution extends Disposable implements IWorkbenchContribution {
-
+	static EXTENSION_RECOMMENDATION_KEY = 'EXTENSION_RECOMMENDATION_KEY';
 	private readonly container: ViewContainer;
 
 	constructor(
 		@IExtensionManagementServerService private readonly extensionManagementServerService: IExtensionManagementServerService,
 		@ILabelService private readonly labelService: ILabelService,
 		@IViewDescriptorService viewDescriptorService: IViewDescriptorService,
-		@IContextKeyService private readonly contextKeyService: IContextKeyService
+		@IContextKeyService private readonly contextKeyService: IContextKeyService,
+		@IProductService private readonly _productService: IProductService,
 	) {
 		super();
 
 		this.container = viewDescriptorService.getViewContainerById(VIEWLET_ID)!;
 		this.registerViews();
+
+		this.contextKeyService.createKey(ExtensionsViewletViewsContribution.EXTENSION_RECOMMENDATION_KEY, !!this._productService.extensionsGallery?.recommendationsUrl);
 	}
 
 	private registerViews(): void {
@@ -238,7 +243,7 @@ export class ExtensionsViewletViewsContr
 			id: 'extensions.recommendedList',
 			name: localize2('recommendedExtensions', "Recommended"),
 			ctorDescriptor: new SyncDescriptor(DefaultRecommendedExtensionsView, [{ flexibleHeight: true }]),
-			when: ContextKeyExpr.and(DefaultViewsContext, SortByUpdateDateContext.negate(), ContextKeyExpr.not('config.extensions.showRecommendationsOnlyOnDemand'), CONTEXT_HAS_GALLERY),
+			when: ContextKeyExpr.and(DefaultViewsContext, SortByUpdateDateContext.negate(), ContextKeyExpr.not('config.extensions.showRecommendationsOnlyOnDemand'), CONTEXT_HAS_GALLERY, ContextKeyExpr.has(ExtensionsViewletViewsContribution.EXTENSION_RECOMMENDATION_KEY)),
 			weight: 40,
 			order: 3,
 			canToggleVisibility: true
@@ -516,7 +521,8 @@ export class ExtensionsViewPaneContainer
 		@IExtensionService extensionService: IExtensionService,
 		@IViewDescriptorService viewDescriptorService: IViewDescriptorService,
 		@IPreferencesService private readonly preferencesService: IPreferencesService,
-		@ICommandService private readonly commandService: ICommandService
+		@ICommandService private readonly commandService: ICommandService,
+		@IProductService private readonly _productService: IProductService,
 	) {
 		super(VIEWLET_ID, { mergeViewWithContainerWhenSingleView: true }, instantiationService, configurationService, layoutService, contextMenuService, telemetryService, extensionService, themeService, storageService, contextService, viewDescriptorService);
 
@@ -544,6 +550,16 @@ export class ExtensionsViewPaneContainer
 		this.searchViewletState = this.getMemento(StorageScope.WORKSPACE, StorageTarget.MACHINE);
 	}
 
+	@memoize
+	get marketplaceHostname(): string {
+
+		if (this._productService.extensionsGallery?.serviceUrl) {
+			return new URL(this._productService.extensionsGallery?.serviceUrl).hostname;
+		}
+
+		return 'Marketplace';
+	}
+
 	get searchValue(): string | undefined {
 		return this.searchBox?.getValue();
 	}
@@ -558,8 +574,7 @@ export class ExtensionsViewPaneContainer
 		hide(overlay);
 
 		const header = append(this.root, $('.header'));
-		const placeholder = localize('searchExtensions', "Search Extensions in Marketplace");
-
+		const placeholder = localize('searchExtensions', 'Search Extensions on {0}', this.marketplaceHostname);
 		const searchValue = this.searchViewletState['query.value'] ? this.searchViewletState['query.value'] : '';
 
 		const searchContainer = append(header, $('.extensions-search-container'));
@@ -924,4 +939,4 @@ export class MaliciousExtensionChecker i
 			}).then(() => undefined);
 		}, err => this.logService.error(err));
 	}
-}
+}
\ No newline at end of file
