Index: sagemaker-code-editor/vscode/src/vs/platform/product/common/marketplace.ts
===================================================================
--- /dev/null
+++ sagemaker-code-editor/vscode/src/vs/platform/product/common/marketplace.ts
@@ -0,0 +1,28 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Coder Technologies. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import { IProductConfiguration } from 'vs/base/common/product';
+
+/**
+ * Allows for overriding the extension gallery configuration via the
+ * `'EXTENSIONS_GALLERY'` environment variable.
+ *
+ * @example
+ * ```sh
+ * export EXTENSIONS_GALLERY='{"serviceUrl": "https://extensions.coder.com/api"}'
+ * ```
+ */
+export function parseExtensionsGalleryEnv(extensionsGalleryEnv: string): IProductConfiguration['extensionsGallery'] {
+    return {
+        serviceUrl: "",
+        itemUrl: "",
+        resourceUrlTemplate: "",
+        controlUrl: '',
+        recommendationsUrl: "",
+        nlsBaseUrl: "",
+        publisherUrl: "", 
+        ...JSON.parse(extensionsGalleryEnv),
+    };
+}
\ No newline at end of file
Index: sagemaker-code-editor/vscode/src/vs/platform/product/common/product.ts
===================================================================
--- sagemaker-code-editor.orig/vscode/src/vs/platform/product/common/product.ts
+++ sagemaker-code-editor/vscode/src/vs/platform/product/common/product.ts
@@ -6,6 +6,7 @@
 import { env } from 'vs/base/common/process';
 import { IProductConfiguration } from 'vs/base/common/product';
 import { ISandboxConfiguration } from 'vs/base/parts/sandbox/common/sandboxTypes';
+import { parseExtensionsGalleryEnv } from 'vs/platform/product/common/marketplace';
 
 /**
  * @deprecated You MUST use `IProductService` if possible.
@@ -74,6 +75,25 @@ else {
 	}
 }
 
+if (typeof env['EXTENSIONS_GALLERY'] !== 'undefined') {
+	console.log(`Default marketplace.`);
+	console.log(JSON.stringify(product.extensionsGallery, null, 2));
+	
+	console.log(`Custom marketplace env found. Parsing...`);
+
+	try {
+		Object.assign(product, {
+			extensionsGallery: parseExtensionsGalleryEnv(env['EXTENSIONS_GALLERY'])
+		});
+	} catch (error) {
+		console.error(error);
+		console.info('Check that your env var is valid JSON and conforms to `IProductConfiguration[\'extensionsGallery\']`>');
+	}
+
+	console.log(`Custom marketplace enabled.`);
+	console.log(JSON.stringify(product.extensionsGallery, null, 2));
+}
+
 /**
  * @deprecated You MUST use `IProductService` if possible.
  */
Index: sagemaker-code-editor/vscode/src/vs/workbench/contrib/extensions/browser/extensionsViewlet.ts
===================================================================
--- sagemaker-code-editor.orig/vscode/src/vs/workbench/contrib/extensions/browser/extensionsViewlet.ts
+++ sagemaker-code-editor/vscode/src/vs/workbench/contrib/extensions/browser/extensionsViewlet.ts
@@ -64,6 +64,8 @@ import { ILocalizedString } from 'vs/pla
 import { registerNavigableContainer } from 'vs/workbench/browser/actions/widgetNavigationCommands';
 import { MenuWorkbenchToolBar } from 'vs/platform/actions/browser/toolbar';
 import { createActionViewItem } from 'vs/platform/actions/browser/menuEntryActionViewItem';
+import { IProductService } from 'vs/platform/product/common/productService';
+import { memoize } from 'vs/base/common/decorators';
 
 export const DefaultViewsContext = new RawContextKey<boolean>('defaultExtensionViews', true);
 export const ExtensionsSortByContext = new RawContextKey<string>('extensionsSortByValue', '');
@@ -87,7 +89,6 @@ const SortByUpdateDateContext = new RawC
 const REMOTE_CATEGORY: ILocalizedString = localize2({ key: 'remote', comment: ['Remote as in remote machine'] }, "Remote");
 
 export class ExtensionsViewletViewsContribution extends Disposable implements IWorkbenchContribution {
-
 	private readonly container: ViewContainer;
 
 	constructor(
@@ -516,7 +517,8 @@ export class ExtensionsViewPaneContainer
 		@IExtensionService extensionService: IExtensionService,
 		@IViewDescriptorService viewDescriptorService: IViewDescriptorService,
 		@IPreferencesService private readonly preferencesService: IPreferencesService,
-		@ICommandService private readonly commandService: ICommandService
+		@ICommandService private readonly commandService: ICommandService,
+		@IProductService private readonly productService: IProductService,
 	) {
 		super(VIEWLET_ID, { mergeViewWithContainerWhenSingleView: true }, instantiationService, configurationService, layoutService, contextMenuService, telemetryService, extensionService, themeService, storageService, contextService, viewDescriptorService);
 
@@ -544,6 +546,15 @@ export class ExtensionsViewPaneContainer
 		this.searchViewletState = this.getMemento(StorageScope.WORKSPACE, StorageTarget.MACHINE);
 	}
 
+	@memoize
+	get marketplaceHostname(): string {
+		if (this.productService.extensionsGallery?.serviceUrl) {
+			return new URL(this.productService.extensionsGallery?.serviceUrl).hostname;
+		}
+
+		return 'Marketplace';
+	}
+
 	get searchValue(): string | undefined {
 		return this.searchBox?.getValue();
 	}
@@ -558,8 +569,7 @@ export class ExtensionsViewPaneContainer
 		hide(overlay);
 
 		const header = append(this.root, $('.header'));
-		const placeholder = localize('searchExtensions', "Search Extensions in Marketplace");
-
+		const placeholder = localize('searchExtensions', 'Search Extensions on {0}', this.marketplaceHostname);
 		const searchValue = this.searchViewletState['query.value'] ? this.searchViewletState['query.value'] : '';
 
 		const searchContainer = append(header, $('.extensions-search-container'));
@@ -924,4 +934,4 @@ export class MaliciousExtensionChecker i
 			}).then(() => undefined);
 		}, err => this.logService.error(err));
 	}
-}
+}
\ No newline at end of file
